File: uikit/src/main/java/utils/PreferenceUtil.java
package utils;

import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceUtil {

    private static final String PREFERENCE_NAME = "UI KIT";

    private static PreferenceUtil INSTANCE=null;

    private static Context context;

    private static SharedPreferences sharedPreferences;

    private PreferenceUtil(Context appContext) {
        context = appContext;
    }

    public static PreferenceUtil getInstance(Context appContext) {

         if (INSTANCE==null)
             INSTANCE=new PreferenceUtil(appContext);

         return INSTANCE;
    }

    public void saveString(String key,String value) {
        sharedPreferences = context.getSharedPreferences(PREFERENCE_NAME, 0);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public boolean saveStringWithConfirmation(String key,String value){
        sharedPreferences = context.getSharedPreferences(PREFERENCE_NAME, 0);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
       return editor.commit();
    }

    public String getString(String key) {
        sharedPreferences = context.getSharedPreferences(PREFERENCE_NAME, 0);
        return sharedPreferences.getString(key, "");
    }

}

File: uikit-kotlin/src/main/java/utils/PreferenceUtil.kt
package utils

import android.content.Context
import android.content.SharedPreferences

public class PreferenceUtil {
    private var sharedPreference: SharedPreferences? = null
    constructor(appContext: Context) {
        context = appContext
    }

    companion object {
        private val PREFERENCE_NAME = "UI KIT"

        private var INSTANCE: PreferenceUtil? = null



        private var context: Context? = null

        fun getInstance(appContext: Context?): PreferenceUtil? {
            if (INSTANCE == null) INSTANCE = PreferenceUtil(appContext!!)
            return INSTANCE
        }



    }
    fun saveStringWithConfirmation(key: String?, value: String?): Boolean {

        sharedPreference = context!!.getSharedPreferences(PREFERENCE_NAME, 0)
        var sharedPreferences = sharedPreference!!
        val editor: SharedPreferences.Editor = sharedPreferences.edit()
        editor.putString(key, value)
        return editor.commit()
    }
    fun saveString(key: String?, value: String?) {
        sharedPreference = context!!.getSharedPreferences(PREFERENCE_NAME, 0)
        var sharedPreferences = sharedPreference!!
        val editor: SharedPreferences.Editor = sharedPreferences.edit()
        editor.putString(key, value)
        editor.apply()
    }
    fun getString(key: String?): String? {
        sharedPreference = context!!.getSharedPreferences(PREFERENCE_NAME, 0)
        var sharedPreferences = sharedPreference!!
        return sharedPreferences.getString(key, "")
    }

}